Stock.playground/Contents.swift

```diff
@@ -3,116 +3,147 @@ import Foundation
+    private func valueAt(_ price: Double, sharesAccrued: Double) -> Double
     {
-        return self.valueAfter(self.monthsSince(date), atPrice: atPrice)
+        let shareValue = price - strikePrice
+        let value = max(0, shareValue)
+        let shareCount = max(0, min(Double(shares), sharesAccrued))
+        return value * shareCount
     }
+}
+
 
-    func valuesBy(dates: [GrantDate], atPrice: Double) -> [Double]
+extension Date
+{
+    static func grantDate(month: Int, day: Int, year: Int) -> Date
     {
-        return dates.map({ self.valueBy($0, atPrice: atPrice) })
+        let signing = NSDateComponents.init()
+        signing.calendar = Calendar.grantDateCalendar()
+        signing.month = month
+        signing.day = day
+        signing.year = year
+        return signing.date!
     }
 
-    func curriedValueBy(date: GrantDate) -> (Double) -> Double
+    func monthsSince(_ date: Date) -> Int
     {
-        return { price in return self.valueAfter(self.monthsSince(date), atPrice: price) }
+        let dateComponents = Calendar.grantDateCalendar().components(.month, from: date, to: self)
+        return dateComponents.month!
     }
 
-    func curriedValueAt(price: Double) -> (GrantDate) -> Double
+    func after(months: Int) -> Date
     {
-        return { date in return self.valueAfter(self.monthsSince(date), atPrice: price) }
+        return after(months, unit: .month)
     }
-}
-
 
-let signing = StockGrant.GrantDate(month: 1, day: 1, year: 2016)
-let signingGrant = StockGrant(shares: 20000, grantDate: signing, cliffMonths: 12, endMonths: 48, strikePrice: 1.22)
+    func after(years: Int) -> Date
+    {
+        return after(years, unit: .year)
+    }
 
-// Two years at $22 (using months or StockGrant.GrantDate)
-let twoYears = StockGrant.GrantDate(month: 1, day: 1, year: 2018)
-signingGrant.valueBy(twoYears, atPrice: 22)
-signingGrant.valueAfter(24, atPrice: 22)
+    private func after(_ amount: Int, unit: Calendar.Unit) -> Date
+    {
+        return Calendar.grantDateCalendar().date(byAdding: unit, value: amount, to: self)!
+    }
+}
 
-// Strike price higher than hypothetical price
-signingGrant.valueBy(twoYears, atPrice: 0.22)
 
-// Vesting cliff not yet hit
-let sixMonths = StockGrant.GrantDate(month: 6, day: 1, year: 2016)
-signingGrant.valueBy(sixMonths, atPrice: 2.00)
+extension Calendar
+{
+    static func grantDateCalendar() -> Calendar
+    {
+        return Calendar(calendarIdentifier: .gregorian)!
+    }
+}
 
-// Vesting schedule completed
-let fiveYears = StockGrant.GrantDate(month: 1, day: 1, year: 2021)
-signingGrant.valueBy(fiveYears, atPrice: 2.00)
 
-// Currying (date)
-let twoYearsCurried = signingGrant.curriedValueBy(twoYears)
-twoYearsCurried(11)
-twoYearsCurried(22)
+let signing = Date.grantDate(month: 1, day: 1, year: 2016)
+let signingGrant = StockGrant(shares: 20000, grantDate: signing, cliffMonths: 12, endMonths: 48, strikePrice: 1.22)
```

Comment:
😅 don't make the new hires jealous.

Also, get fancy and use `20,000` rather than `20000`